packages/runtime/src/useCases/transport/relationshipTemplates/CreateOwnRelationshipTemplate.ts

```diff
@@ -39,6 +40,17 @@ class Validator extends SchemaValidator<CreateOwnRelationshipTemplateRequest> {
             );
         }
 
+        if (input.password && /^\d+$/.test(input.password) && (input.password.length > 12 || input.password.length < 2)) {
+            validationResult.addFailure(
+                new ValidationFailure(
+                    RuntimeErrors.general.invalidPropertyValue(
+                        `Your chosen '${nameof<CreateOwnRelationshipTemplateRequest>((r) => r.password)}' is a PIN (consists of numbers only) and PINs must be at least 2 and at most 12 digits long`
+                    ),
+                    nameof<CreateOwnRelationshipTemplateRequest>((r) => r.password)
+                )
+            );
+        }
```

Comment:
plase put this to a private method to be able to do early return :)

Reply:
What do you mean? This is part of the UseCase validation, what's supposed to be the improvement?

Reply:
readability.

Reply:
a bunch of boolean logic stuffed in one line can be overwhelming if not written yourself.

Reply:
The minimum PIN length was for the technical reason that we wanted passwordType 1 for non-PINs. Using passwordType 0 for non-PINs would solve that. The PIN characterization was always just for app displaying purposes which shouldn't depend on the length, then there would just be no upper bound on the passwordType.

I  don't know about max or min length - iirc the upper bound on the PIN was because at that point it would be user-friendlier to also use letters. IMO people should know what length constitutes a secure password/PIN for their use and we don't need to enforce it.

Reply:
^ with the above information the password `1234567891011121314` must be of type `1`, you just throw instead.

Reply:
I'm ok with that.

Reply:
And regarding the lower bound? I'd be happy with a lower bound for the length (e. g. 2) as well as allowing 1-digit PINs that are then not treated as passwords :)

Reply:
Should the bound be extended to passwords? Seems like it would be easier for us

Reply:
I think it's way easier to  brute-force if you have the length, so I would avoid that.

Reply:
You would only know that the length is at least 2